sources/representation/qt/mainform.cpp

```diff
@@ -178,43 +184,95 @@ void MainForm::startGameLoop(int id, QString map)
 
     QElapsedTimer process_performance;
     qint64 max_process_time = 0;
-
-    while (true)
+    
+    if (!cap)
```

Comment:
This `if` seems really unnecessary. You may check for it inside the `while(true)` loop.